reconcile/test/test_utils_terraform_client.py

```diff
@@ -56,3 +57,99 @@ def test_expiration_value_error(aws_api):
     tf = tfclient.TerraformClient("integ", "v1", "integ_pfx", [account], {}, 1, aws_api)
     with pytest.raises(tfclient.DeletionApprovalExpirationValueError):
         tf.deletion_approved("a1", "t1", "n1")
+
+
+def test_get_replicas_info_via_replica_source():
+    namespace = {
+        "terraformResources": [
+            {
+                "account": "acc",
+                "identifier": "replica-id",
+                "provider": "rds",
+                "defaults": "defaults-ref",
+                "replica_source": "replica-source-id",
+            }
+        ]
+    }
+    result = tfclient.TerraformClient.get_replicas_info([namespace])
+    expected = {"acc": {"replica-id-rds": "replica-source-id-rds"}}
+    assert result == expected
+
+
+def test_get_replicas_info_via_replica_source_overrides_present():
+    # this test shows that the direct replica_source on the tf resource
+    # has precendence over overrides
+    namespace = {
```

Comment:
Wouldn't it be better to convert these `namespace` declarations into a fixture, like:

```python
@pytest.fixture
def namespace():
   return {
     "terraformResources": [
              ...
     ]
   }


def test_get_replicas_info_via_replica_source_overrides_present(namespace):
     result = tfclient.TerraformClient....

def test_get_replicas_info_via_defaults(mocker, namespace):
    del namespace["terraformResources"][0]["replica_source"]
    result =...

...
```

Reply:
i've chosen the approach of having the "fixture" defined within the test because it makes reviewing them easier. the entire test with all data is in this case small enough to fit on a small part of the screen and i don't have to think about how the real fixture looks like.

i know i'm repeating myself a bit with redeclaring very similar testdata quite some time. it is always a fine line between DRY and DAMP.

i will keep fixture in mind though for other tests